### PR TITLE
musig-spec: add Session Context to simplify sign/verify/sigagg

### PR DESCRIPTION
**Pull request by jonasnick**

### CODE REVIEW

File: doc/musig-spec.mediawiki [hunk 1]
```diff
@@ -147,35 +149,51 @@ The algorithm ''NonceAgg(pubnonce<sub>1..u</sub>)'' is defined as:
...
-* The secret key ''sk'': a 32-byte array
+The Session Context is a data structure consisting of the following elements:
 * The aggregate public nonce ''aggnonce'': a 66-byte array
 * The number ''u'' of public keys with ''0 < u < 2^32''
 * The public keys ''pk<sub>1..u</sub>'': ''u'' 32-byte arrays
 * The message ''m'': a 32-byte array
 
-The algorithm ''Sign(secnonce, sk, aggnonce, pk<sub>1..u</sub>, m)'' is defined as:
-* Let ''R<sub>1</sub> = pointc(aggnonce[0:33]), R<sub>2</sub> = pointc(aggnonce[33:66])''; fail if that fails
+We write "Let ''(aggnonce, u, pk<sub>1..u</sub>, m) = session_ctx''" to assign names to the the elements of a Session Context.
```
Comment (robot-dreams) [inline doc/musig-spec.mediawiki:160]:
```suggestion
We write "Let ''(aggnonce, u, pk<sub>1..u</sub>, m) = session_ctx''" to assign names to the elements of a Session Context.
```

**Review by robot-dreams — COMMENTED** [review]
Just one nit, otherwise looks good.

See discussion in https://github.com/ElementsProject/secp256k1-zkp/pull/167#pullrequestreview-900217911 for previous context. I also pushed #166 (I'll try to keep the branch in sync with spec updates).

**Comment by robot-dreams** [comment]
ACK 5576a9efcbb24871c88b24cf3b9afabd43bb489f

**Comment by jonasnick** [comment]
Closing this because discussions on #167 also affect commits in this PR, so better to focus on #167.

**Pull request closed without merging**